### ISSUE TITLE
Top crashes in v2.8.4

### ISSUE THREAD

**Issue by misaochan** [commons-app/apps-android-commons#1917]
Top crashes in our Play Store developer console currently. Luckily most of these are NPEs which should be quite simple to fix. :) If anyone is interested in working on this, please tell us which crash exactly you are working on so that we don't duplicate effort.

1. java.lang.NullPointerException at fr.free.nrw.commons.explore.images.SearchImageFragment.updateImageList

```
io.reactivex.exceptions.OnErrorNotImplementedException: 
  at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept (Functions.java:704)
  at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept (Functions.java:701)
  at io.reactivex.internal.observers.LambdaObserver.onError (LambdaObserver.java:74)
  at io.reactivex.internal.observers.LambdaObserver.onNext (LambdaObserver.java:64)
  at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.drainNormal (ObservableObserveOn.java:200)
  at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.run (ObservableObserveOn.java:252)
  at io.reactivex.android.schedulers.HandlerScheduler$ScheduledRunnable.run (HandlerScheduler.java:109)
  at android.os.Handler.handleCallback (Handler.java:754)
  at android.os.Handler.dispatchMessage (Handler.java:95)
  at android.os.Looper.loop (Looper.java:163)
  at android.app.ActivityThread.main (ActivityThread.java:6228)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:904)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:794)
Caused by: java.lang.NullPointerException: 
  at fr.free.nrw.commons.explore.images.SearchImageFragment.updateImageList (SearchImageFragment.java:126)
  at fr.free.nrw.commons.explore.SearchActivity.lambda$setTabs$1$SearchActivity (SearchActivity.java:114)
  at fr.free.nrw.commons.explore.SearchActivity$$Lambda$1.accept (Unknown Source)
  at io.reactivex.internal.observers.LambdaObserver.onNext (LambdaObserver.java:60)
```

2. java.lang.NullPointerException at fr.free.nrw.commons.utils.FileUtils.saveFileFromURI

```
java.lang.RuntimeException: 
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:2793)
  at android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:2864)
  at android.app.ActivityThread.-wrap12 (ActivityThread.java)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1567)
  at android.os.Handler.dispatchMessage (Handler.java:105)
  at android.os.Looper.loop (Looper.java:156)
  at android.app.ActivityThread.main (ActivityThread.java:6523)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:942)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:832)
Caused by: java.lang.NullPointerException: 
  at fr.free.nrw.commons.utils.FileUtils.saveFileFromURI (FileUtils.java:52)
  at fr.free.nrw.commons.utils.ContributionUtils.saveFileBeingUploadedTemporarily (ContributionUtils.java:35)
  at fr.free.nrw.commons.upload.ShareActivity.receiveImageIntent (ShareActivity.java:355)
  at fr.free.nrw.commons.upload.ShareActivity.onCreate (ShareActivity.java:302)
  at android.app.Activity.performCreate (Activity.java:6915)
  at android.app.Instrumentation.callActivityOnCreate (Instrumentation.java:1123)
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:2746)

```

3. android.content.res.Resources$NotFoundException at fr.free.nrw.commons.contributions.ContributionsActivity.displayUploadCount

```
io.reactivex.exceptions.UndeliverableException: 
  at io.reactivex.plugins.RxJavaPlugins.onError (RxJavaPlugins.java:349)
  at io.reactivex.internal.observers.ConsumerSingleObserver.onSuccess (ConsumerSingleObserver.java:64)
  at io.reactivex.internal.operators.single.SingleObserveOn$ObserveOnSingleObserver.run (SingleObserveOn.java:81)
  at io.reactivex.android.schedulers.HandlerScheduler$ScheduledRunnable.run (HandlerScheduler.java:109)
  at android.os.Handler.handleCallback (Handler.java:789)
  at android.os.Handler.dispatchMessage (Handler.java:98)
  at android.os.Looper.loop (Looper.java:164)
  at android.app.ActivityThread.main (ActivityThread.java:6940)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.Zygote$MethodAndArgsCaller.run (Zygote.java:327)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1374)
Caused by: android.content.res.Resources$NotFoundException: 
  at android.content.res.ResourcesImpl.getQuantityText (ResourcesImpl.java:294)
  at android.content.res.Resources.getQuantityText (Resources.java:428)
  at android.content.res.Resources.getQuantityString (Resources.java:504)
  at fr.free.nrw.commons.contributions.ContributionsActivity.displayUploadCount (ContributionsActivity.java:306)
  at fr.free.nrw.commons.contributions.ContributionsActivity.bridge$lambda$0$ContributionsActivity (Unknown Source)
  at fr.free.nrw.commons.contributions.ContributionsActivity$$Lambda$0.accept (Unknown Source:4)
  at io.reactivex.internal.observers.ConsumerSingleObserver.onSuccess (ConsumerSingleObserver.java:61)
```

4. java.lang.NullPointerException at fr.free.nrw.commons.location.LatLng.from

```
java.lang.RuntimeException: 
  at android.app.ActivityThread.deliverResults (ActivityThread.java:4517)
  at android.app.ActivityThread.handleSendResult (ActivityThread.java:4560)
  at android.app.ActivityThread.-wrap19 (Unknown Source)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1744)
  at android.os.Handler.dispatchMessage (Handler.java:105)
  at android.os.Looper.loop (Looper.java:164)
  at android.app.ActivityThread.main (ActivityThread.java:6798)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.Zygote$MethodAndArgsCaller.run (Zygote.java:240)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:767)
Caused by: java.lang.NullPointerException: 
  at fr.free.nrw.commons.location.LatLng.from (LatLng.java:45)
  at fr.free.nrw.commons.location.LocationServiceManager.getLKL (LocationServiceManager.java:109)
  at fr.free.nrw.commons.nearby.NearbyActivity.onRequestPermissionsResult (NearbyActivity.java:224)
  at android.app.Activity.dispatchRequestPermissionsResult (Activity.java:7425)
  at android.app.Activity.dispatchActivityResult (Activity.java:7276)
  at android.app.ActivityThread.deliverResults (ActivityThread.java:4513)
```

5.  java.lang.NullPointerException at fr.free.nrw.commons.category.CategoryImagesListFragment.getAdapter

```
java.lang.RuntimeException: 
  at android.app.ActivityThread.performResumeActivity (ActivityThread.java:3715)
  at android.app.ActivityThread.handleResumeActivity (ActivityThread.java:3755)
  at android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:2939)
  at android.app.ActivityThread.handleRelaunchActivity (ActivityThread.java:4833)
  at android.app.ActivityThread.-wrap18 (Unknown Source)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1618)
  at android.os.Handler.dispatchMessage (Handler.java:105)
  at android.os.Looper.loop (Looper.java:164)
  at android.app.ActivityThread.main (ActivityThread.java:6710)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.Zygote$MethodAndArgsCaller.run (Zygote.java:240)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:770)
Caused by: java.lang.NullPointerException: 
  at fr.free.nrw.commons.category.CategoryImagesListFragment.getAdapter (CategoryImagesListFragment.java:267)
  at fr.free.nrw.commons.category.CategoryDetailsActivity.getMediaAtPosition (CategoryDetailsActivity.java:156)
  at fr.free.nrw.commons.media.MediaDetailFragment.onResume (MediaDetailFragment.java:207)
  at android.support.v4.app.Fragment.performResume (Fragment.java:2390)
  at android.support.v4.app.FragmentManagerImpl.moveToState (FragmentManager.java:1474)
  at android.support.v4.app.FragmentManagerImpl.moveFragmentToExpectedState (FragmentManager.java:1759)
  at android.support.v4.app.FragmentManagerImpl.moveToState (FragmentManager.java:1827)
  at android.support.v4.app.FragmentManagerImpl.dispatchStateChange (FragmentManager.java:3244)
  at android.support.v4.app.FragmentManagerImpl.dispatchResume (FragmentManager.java:3212)
  at android.support.v4.app.Fragment.performResume (Fragment.java:2396)
  at android.support.v4.app.FragmentManagerImpl.moveToState (FragmentManager.java:1474)
  at android.support.v4.app.FragmentManagerImpl.moveFragmentToExpectedState (FragmentManager.java:1759)
  at android.support.v4.app.FragmentManagerImpl.moveToState (FragmentManager.java:1827)
  at android.support.v4.app.FragmentManagerImpl.dispatchStateChange (FragmentManager.java:3244)
  at android.support.v4.app.FragmentManagerImpl.dispatchResume (FragmentManager.java:3212)
  at android.support.v4.app.FragmentController.dispatchResume (FragmentController.java:217)
  at android.support.v4.app.FragmentActivity.onResumeFragments (FragmentActivity.java:509)
  at android.support.v4.app.FragmentActivity.onPostResume (FragmentActivity.java:498)
  at android.support.v7.app.AppCompatActivity.onPostResume (AppCompatActivity.java:171)
  at android.app.Activity.performResume (Activity.java:7110)
  at android.app.ActivityThread.performResumeActivity (ActivityThread.java:3690)
```

6.  java.lang.NullPointerException at fr.free.nrw.commons.category.CategoryImagesListFragment.getAdapter

```
java.lang.RuntimeException: 
  at android.app.ActivityThread.performResumeActivity (ActivityThread.java:4035)
  at android.app.ActivityThread.handleResumeActivity (ActivityThread.java:4100)
  at android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:3271)
  at android.app.ActivityThread.handleRelaunchActivity (ActivityThread.java:5200)
  at android.app.ActivityThread.access$1100 (ActivityThread.java:218)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1740)
  at android.os.Handler.dispatchMessage (Handler.java:102)
  at android.os.Looper.loop (Looper.java:145)
  at android.app.ActivityThread.main (ActivityThread.java:6934)
  at java.lang.reflect.Method.invoke (Native Method)
  at java.lang.reflect.Method.invoke (Method.java:372)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1404)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1199)
Caused by: java.lang.NullPointerException: 
  at fr.free.nrw.commons.category.CategoryImagesListFragment.getAdapter (CategoryImagesListFragment.java:267)
  at fr.free.nrw.commons.category.CategoryDetailsActivity.getMediaAtPosition (CategoryDetailsActivity.java:156)
  at fr.free.nrw.commons.media.MediaDetailFragment.onResume (MediaDetailFragment.java:207)
  at android.support.v4.app.Fragment.performResume (Fragment.java:2390)
  at android.support.v4.app.FragmentManagerImpl.moveToState (FragmentManager.java:1474)
  at android.support.v4.app.FragmentManagerImpl.moveFragmentToExpectedState (FragmentManager.java:1759)
  at android.support.v4.app.FragmentManagerImpl.moveToState (FragmentManager.java:1827)
  at android.support.v4.app.FragmentManagerImpl.dispatchStateChange (FragmentManager.java:3244)
  at android.support.v4.app.FragmentManagerImpl.dispatchResume (FragmentManager.java:3212)
  at android.support.v4.app.Fragment.performResume (Fragment.java:2396)
  at android.support.v4.app.FragmentManagerImpl.moveToState (FragmentManager.java:1474)
  at android.support.v4.app.FragmentManagerImpl.moveFragmentToExpectedState (FragmentManager.java:1759)
  at android.support.v4.app.FragmentManagerImpl.moveToState (FragmentManager.java:1827)
  at android.support.v4.app.FragmentManagerImpl.dispatchStateChange (FragmentManager.java:3244)
  at android.support.v4.app.FragmentManagerImpl.dispatchResume (FragmentManager.java:3212)
  at android.support.v4.app.FragmentController.dispatchResume (FragmentController.java:217)
  at android.support.v4.app.FragmentActivity.onResumeFragments (FragmentActivity.java:509)
  at android.support.v4.app.FragmentActivity.onPostResume (FragmentActivity.java:498)
  at android.support.v7.app.AppCompatActivity.onPostResume (AppCompatActivity.java:171)
  at android.app.Activity.performResume (Activity.java:6748)
  at android.app.ActivityThread.performResumeActivity (ActivityThread.java:4024)
```

**Comment by neslihanturan** [comment]
Seems like we had a successful release!:)

**Comment by misaochan** [comment]
@neslihanturan Yes, the crash rates have dropped a lot! :D We still have a few other issues like images being queued, but one step at a time...

**Comment by henrique-gb** [comment]
Hello. I'm new to opensource. I'd like to work on this issue.
By the way, crashes 5 and 6 look the same to me. Is this really the case?

**Comment by misaochan** [comment]
@henrique-gb Ah, you are right actually! Would you like to work on crash 5/6, then? We can assign you other crashes once you have solved that one. :)

**Comment by henrique-gb** [comment]
@misaochan I think I was able to fix crashes 1, 2 and 5/6 by just adding simple null checks.
I'm having some trouble reproducing crashes 3 and 4.
As far as I can tell, b73535f62136eaeb61fe18b88e2c3f35a9ff185d should prevent crash 4 from happening.

**Comment by misaochan** [comment]
@henrique-gb Great! :) Unfortunately I think crash 4 is still happening with the commit you mentioned, as it is tagged in v2.8.4 but we are still getting that crash in that version. But if you can submit your PR with all of the others, that would still be good.

**Comment by svendhhh** [comment]
Are 3 and/or 4 being looked at already? I had a quick look at 4, and I don't understand how it could happen, as it would require a @NotNull parameter being null... unless it's just incorrectly annotated?

**Comment by hoplite390** [comment]
@misaochan number 4 is probably solved in master but not in 2.8 branch, because i cant recreate it with master. It seems that it occurs with app-store version and 2.8 branch only. (I did push request to 2.8 however if i understand correctly it seems that 2.8 branch was merged to master and it will not be developed anymore.) Please correct me if that is not the case.

**Comment by misaochan** [comment]
Thanks for the insight @hoplite390 ! Yes, we don't intend to release any further 2.8 versions, unless a very urgent bug is discovered, and indeed the 2.8 branch was merged to master. If it is fixed in master, then that should be fine.
